public/app/dashboard/activityNetworkDirective.js

```diff
@@ -329,22 +341,53 @@
           };
 
           // Size force diagram to window, accounting for course size and resizing as necessary.
+          var activityNetworkScale = Math.round(Math.sqrt(4000 * scope.interactions.nodes.length));
+          var bounds;
           var viewportWidth;
           var viewportHeight;
-          var activityNetworkScale = Math.round(Math.sqrt(1000 * scope.interactions.nodes.length));
+
+          scope.minScale = 1;
 
           var sizeAndRestart = function() {
             viewportWidth = controlsForm.node().getBoundingClientRect().width;
-            viewportHeight = 400;
-            svg.attr('width', viewportWidth).attr('height', 300);
-            clipPathRect.attr('width', viewportWidth).attr('height', 300);
-            simulation.force('charge', d3.forceManyBody().strength(-0.6 * activityNetworkScale).distanceMax(300));
+            viewportHeight = Math.max(400, Math.min(250, activityNetworkScale));
+
+            var aspectRatio = viewportWidth * 1.0 / viewportHeight;
+
+            svg.attr('width', viewportWidth).attr('height', viewportHeight);
+            clipPathRect.attr('width', viewportWidth).attr('height', viewportHeight);
+
+            var xMargin = Math.max(0, (Math.round(activityNetworkScale * aspectRatio) - viewportWidth) / 2);
+            var yMargin = Math.max(0, (activityNetworkScale - viewportHeight) / 2);
+            bounds = [[-1 * xMargin, -1 * yMargin], [viewportWidth + xMargin, viewportHeight + yMargin]];
+            scope.minScale = Math.min(viewportHeight / activityNetworkScale, 1);
+            zoom.scaleExtent([scope.minScale, 1]).traånslateExtent(bounds);
```

Comment:
@paulkerschen Please tell me that 'traånslateExtent' is correct...

Reply:
Hvorfor gjør Sublime Text disse tegnene???